lib/ansible/modules/network/nuage/nuage_vspk.py

```diff
@@ -0,0 +1,1089 @@
+
+        self.api_username = self.auth['api_username']
+        if 'api_password' in list(self.auth.keys()) and self.auth['api_password']:
+            self.api_password = self.auth['api_password']
+        if 'api_certificate' in list(self.auth.keys()) and 'api_key' in list(self.auth.keys()) and self.auth['api_certificate'] and self.auth['api_key']:
+            self.api_certificate = self.auth['api_certificate']
+            self.api_key = self.auth['api_key']
+        self.api_enterprise = self.auth['api_enterprise']
+        self.api_url = self.auth['api_url']
+        self.api_version = self.auth['api_version']
+
+        try:
+            global VSPK
+            VSPK = importlib.import_module('vspk.{0:s}'.format(self.api_version))
+        except ImportError:
+            self.module.fail_json(msg='vspk is required for this module, or the API version specified does not exist.')
+
+    def _verify_input(self):
+        """
+        Verifies the parameter input for types and parent correctness and necessary parameters
+        """
+
+        # Checking if type exists
+        try:
+            self.entity_class = getattr(VSPK, 'NU{0:s}'.format(self.type))
+        except AttributeError:
+            self.module.fail_json(msg='Unrecognised type specified')
+
+        if self.module.check_mode:
+            return
+
+        if self.parent_id and not self.parent_type:
+            # Checking if parent info is ok
+            self.module.fail_json(msg='Parent ID specified, but no parent type specified')
+        elif self.parent_type and not self.parent_id:
+            # Checking if parent info is ok
+            self.module.fail_json(msg='Parent type specified, but no parent id specified')
+        elif self.parent_type:
+            # Checking if parent type exists
+            try:
+                self.parent_class = getattr(VSPK, 'NU{0:s}'.format(self.parent_type))
+            except AttributeError:
+                # The parent type does not exist, fail
+                self.module.fail_json(msg='Unrecognised parent type specified')
+
+            fetcher = self.parent_class().fetcher_for_rest_name(self.entity_class.rest_name)
+            if fetcher is None:
+                # The parent has no fetcher, fail
+                self.module.fail_json(msg='Specified parent is not a valid parent for the specified type')
+        elif not self.entity_id:
+            # If there is an id, we do not need a parent because we'll interact directly with the entity
+            # If an assign needs to happen, a parent will have to be provided
+            # Root object is the parent
+            self.parent_class = VSPK.NUMe
+            fetcher = self.parent_class().fetcher_for_rest_name(self.entity_class.rest_name)
+            if fetcher is None:
+                self.module.fail_json(msg='No parent specified and root object is not a parent for the type')
+
+        # Verifying state or command is set:
```

Comment:
I think most if not all of this can be removed and replaced with argspec code

https://github.com/gundalow/ansible/blob/docs-argspec/docs/docsite/rst/dev_guide/developing_modules_general.rst#main-and-ansiblemodule-argument-spec